File: src/main/java/com/dude/dms/ui/components/search/DmsSearchOverlayButton.java
package com.dude.dms.ui.components.search;

import com.github.appreciated.app.layout.component.appbar.IconButton;
import com.vaadin.flow.component.AttachEvent;
import com.vaadin.flow.component.DetachEvent;
import com.vaadin.flow.component.icon.VaadinIcon;

public class DmsSearchOverlayButton extends IconButton {

    private final DmsSearchOverlayView searchView;

    public DmsSearchOverlayButton() {
        super(VaadinIcon.SEARCH.create());
        searchView = new DmsSearchOverlayView();
        addClickListener(event -> searchView.open());
    }

    @Override
    protected void onDetach(DetachEvent detachEvent) {
        super.onDetach(detachEvent);
        searchView.getElement().removeFromParent();
    }

    @Override
    protected void onAttach(AttachEvent attachEvent) {
        super.onAttach(attachEvent);
        attachEvent.getUI().add(searchView);
    }

    public DmsSearchOverlayView getSearchView() {
        return searchView;
    }
}

File: src/main/kotlin/com/dude/dms/ui/components/search/DmsSearchOverlayButton.kt
package com.dude.dms.ui.components.search

import com.dude.dms.ui.builder.BuilderFactory
import com.github.appreciated.app.layout.component.appbar.IconButton
import com.vaadin.flow.component.AttachEvent
import com.vaadin.flow.component.DetachEvent
import com.vaadin.flow.component.icon.VaadinIcon

class DmsSearchOverlayButton(builderFactory: BuilderFactory) : IconButton(VaadinIcon.SEARCH.create()) {

    val searchView = DmsSearchOverlayView(builderFactory)

    init {
        addClickListener { searchView.open() }
    }

    override fun onAttach(attachEvent: AttachEvent) {
        super.onAttach(attachEvent)
        attachEvent.ui.add(searchView)
    }

    override fun onDetach(detachEvent: DetachEvent) {
        super.onDetach(detachEvent)
        searchView.element.removeFromParent()
    }
}